### ISSUE TITLE
[Bug] Argument of type 'X' is not assignable to parameter of type 'TimelineDefinition'.

### ISSUE THREAD

**Issue by Ladvace** [motiondivision/motionone#178]
**1. Describe the bug**

I was trying to create a simple timeline  as in the example below but I got the following ts error:

`
const sequence: ((string | {
    x: number;
    duration?: undefined;
} | {
    duration: number;
    x?: undefined;
})[] | (string | {
    opacity: number;
    duration?: undefined;
    delay?: undefined;
} | {
    duration: number;
    delay: OptionResolver<number>;
    opacity?: undefined;
})[])[]
Argument of type '((string | { x: number; duration?: undefined; } | { duration: number; x?: undefined; })[] | (string | { opacity: number; duration?: undefined; delay?: undefined; } | { duration: number; delay: OptionResolver<number>; opacity?: undefined; })[])[]' is not assignable to parameter of type 'TimelineDefinition'.
  Type '(string | { x: number; duration?: undefined; } | { duration: number; x?: undefined; })[] | (string | { opacity: number; duration?: undefined; delay?: undefined; } | { duration: number; delay: OptionResolver<number>; opacity?: undefined; })[]' is not assignable to type 'TimelineSegment'.
    Type '(string | { x: number; duration?: undefined; } | { duration: number; x?: undefined; })[]' is not assignable to type 'TimelineSegment'.
      Type '(string | { x: number; duration?: undefined; } | { duration: number; x?: undefined; })[]' is not assignable to type '[ElementOrSelector, MotionKeyframesDefinition, AnimationListOptions]'.
        Target requires 3 element(s) but source may have fewer.ts(2345)`

```
    const sequence = [
      [".title", { x: 100 }, { duration: 1 }],
      [".subTitle", { opacity: 1 }, { duration: 0.3, delay: stagger(0.1) }]
    ];

    timeline(sequence);
```
**2. IMPORTANT: Provide a CodeSandbox reproduction of the bug**

[codesandbox](https://codesandbox.io/s/cranky-mirzakhani-96tn27?file=/src/index.tsx)

**6. Browser details**

chrome

**Comment by legowhales** [comment]
Your example woks perfectly fine on my end.

**Comment by Ladvace** [comment]
> Your example works perfectly fine on my end.

I have attached a codesandbox where  you can see the error

**Comment by legowhales** [comment]
I know, and when I run your codesanbox, everything works as expected (no error at all 🤔).

**Comment by Ladvace** [comment]
strange, I get this:

<img width="643" alt="image" src="https://user-images.githubusercontent.com/31326058/224095193-83039f00-a3e3-492c-9702-1db4f1ec61ff.png">

**Comment by legowhales** [comment]
Oh ok, I thought you had a bug, that the timeline did not animate. This is more a typescript issue then.
Having a way to import the TimelineDefinition type would probably do it. But it does not seem to be exported.
I am not a typescript specialist, maybe @mattgperry could help you there.

Alternately this should work:

```typescript
const sequence = [
  [".title", { x: 100 }, { duration: 1 }],
  [".subTitle", { opacity: 1 }, { duration: 0.3, delay: stagger(0.1) }]
] as any;
timeline(sequence);
```
Or just:
```typescript
timeline([
  [".title", { x: 100 }, { duration: 1 }],
  [".subTitle", { opacity: 1 }, { duration: 0.3, delay: stagger(0.1) }]
]);
```

**Comment by Ladvace** [comment]
yeah sorry for not clarifying, it's just a TS problem, the timeline itself works fine, yes I easily fixed with an "as any", this is more a workaround for that

**Comment by aseeeem** [comment]
Hi, I encountered a the same error but was able to get around it.
I was able to fix it by just adding the `satisfies` keyword where I'm constructing my `sequence` definition.

So with code, it's just:

```ts
import { stagger, timeline, type TimelineDefinition } from "motion";

const sequence = [
  ["p", { opacity: [0.2, 1] }, { delay: stagger(0.9) }],
  [
    "span[data-check-me-out]",
    {
      color: [null, "#ff0000"],
    },
    {
      duration: 1,
      at: "+1",
    },
  ],
] satisfies TimelineDefinition;

timeline(sequence);
```

And no more error! You could do that, or just say `const sequence: TimelineDefinition` too, whatever u want

I think the root of this issue is that when you're constructing the object and passing it to the `timeline` function, TS thinks its just an array of length `n` and not a tuple. 

See the error is:

```
Type '(string | { opacity: number[]; delay?: undefined; } | { delay: OptionResolver<number>; opacity?: undefined; })[]' is not assignable to type '[ElementOrSelector, MotionKeyframesDefinition, AnimationListOptions]'.
```

The error states that the type of `sequence` is an array of an unknown length, whereas the parameter has a discrete length of 3.
Again I'm not the sickest with typescript so there's probably a way within the types to help narrow them down, but I'm unsure.

Either way, I hope this helps.
